### PR TITLE
Remove Declarations.yaml dependency from gen_autograd

### PR DESCRIPTION
**Pull request by peterbell10**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #67497
* __->__ #67496

gen_autograd.py doesn't use `Declarations.yaml` any more, and removing
the dependency allows it to run in parallel with
`tools/codegen/gen.py`.

Differential Revision: [D32027251](https://our.internmc.facebook.com/intern/diff/D32027251)

### CODE REVIEW

**Review by albanD — COMMENTED** [review]
Ho wow great catch!  when did that happen?

**Review by albanD — APPROVED** [review]

**Comment by albanD** [comment]
@albanD has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32027251).

**Comment by peterbell10** [comment]
Yeah, I was expecting this to be a bit more work... Looks like #49735 did all the hard work but didn't update the build system.

File: BUILD.bazel [hunk 1]
```diff
@@ -246,7 +246,7 @@ genrule(
         "aten/src/ATen/native/native_functions.yaml",
```
Comment (dagitses) [inline BUILD.bazel:246]: please remove the dependency from the srcs here

**Review by dagitses — CHANGES_REQUESTED** [review]
looks like you missed the dep on the bazel rule

File: BUILD.bazel (hunk 1 above)
Comment (albanD) [reply BUILD.bazel:246]: Can this be done in a follow up? This just add un-necessary dependency to the bazel build right?

**Comment by albanD** [comment]
Landing failed internally anyways...
@peterbell10 could you do the update suggested above please?

File: BUILD.bazel (hunk 1 above)
Comment (peterbell10) [reply BUILD.bazel:245]: Done

**Comment by albanD** [comment]
Thanks for the quick update!

**Comment by albanD** [comment]
@albanD has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32027251).

File: BUILD.bazel (hunk 1 above)
Comment (dagitses) [reply BUILD.bazel:245]: correct this could have been done in a follow up

**Review by dagitses — APPROVED** [review]

**Pull request closed without merging**